### PR TITLE
Allow superusers to create and remove users

### PR DESCRIPTION
**Pull request by joesingo**
Superusers can now create new users with
`$ mcvirt user create <new username>`
The password for the new user can be provided with `--user-password=<new password>`, generated for you with `--generate-password` (the password is printed out afterwards) or typed in interactively.

Superusers can also remove users with
`$ mcvirt user remove <username>`

### CODE REVIEW

**Comment by itdevreviewboard** [comment]
Can one of the MCVirt admins verify this patch?

File: source/usr/lib/mcvirt/parser.py [hunk 1]
```diff
@@ -125,6 +125,40 @@ def __init__(self, verbose=True, auth_object=None):
...
+            help='Create a new user',
+            parents=[self.parent_parser]
+        )
+        self.create_user_subparser.add_argument(
+            'new_username',
+            metavar='User',
+            type=str,
+            help='The new user to create'
+        )
+        self.create_user_subparser.add_argument(
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/parser.py]: I would create a mutual_exclusive_group here to ensure that --generate-password and --user-password cannot both be passed

File: source/usr/lib/mcvirt/auth/factory.py [hunk 2]
```diff
@@ -43,6 +43,7 @@ def ensure_valid_user_type(self, user_type):
         if user_type not in self.get_user_types():
             raise InvalidUserTypeException('An invalid user type has been passed')
 
+    @Pyro4.expose()
     def create(self, username, password, user_type=User):
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/auth/factory.py]: I would add a check in here to ensure that a password isn't blank

File: source/usr/lib/mcvirt/system.py [hunk 3]
```diff
@@ -52,3 +52,14 @@ def getUserInput(display_text, password=False):
         else:
             sys.stdout.write(display_text)
             return sys.stdin.readline()
+
+    @staticmethod
+    def getNewPassword():
+        """Prompts the user for a new password, throwing an exception is the password is not
```
Comment (MatthewJohn) [inline source/usr/lib/mcvirt/system.py]: sorry to be a pain, pylint will require the comment to be:

    def blah():
        """My multline
        comment
        """

**Comment by MatthewJohn** [comment]
:shipit:

**Pull request merged**